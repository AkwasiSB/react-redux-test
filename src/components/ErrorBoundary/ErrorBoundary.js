import PropTypes from 'prop-types';
import React from 'react';

class ErrorBoundary extends React.PureComponent {
	static propTypes = {
		children: PropTypes.element
	}

	state = {
		isError: false,
	}

	static getDerivedStateFromError() {
		return {
			isError: true
		};
	}

	render() {
		if (!this.state.isError) {
			return this.props.children;
		}

		return (
			<p>Oops! an error occured. Please check your connection and try again :)</p>
		);
	}
}

export default ErrorBoundary;
