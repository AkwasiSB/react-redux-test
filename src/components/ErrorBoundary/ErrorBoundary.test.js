import ErrorBoundary from './ErrorBoundary';

test('if ErrorBoundary is rendered correctly', () => {
    const errorWrapper = shallow(<ErrorBoundary />);

    expect(errorWrapper).toMatchSnapshot();

    errorWrapper.setState({
        isError: true
    });

    expect(errorWrapper.text()).toEqual('Oops! an error occured. Please check your connection and try again :)');
});
