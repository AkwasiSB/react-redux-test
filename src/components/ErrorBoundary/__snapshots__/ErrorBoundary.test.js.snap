// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if ErrorBoundary is rendered correctly 1`] = `""`;
